README.md

```diff
@@ -1,107 +1,207 @@
+     `deno.land/std`, and `nest.land/std`.
+2. Typecheck the code. You want that.
+3. Emit `.js` and `.d.ts` files. These files can directly be run by Node or
+   published on npm.
 
-Note: output and diagnostics will change across minor versions.
+Note that `deno2node` does all of this in memory. It does not touch your source
+files, and only writes the build artefacts to your disk.
+
+`deno2node` uses [`ts-morph`] under the hood, which in turn builds on top of the
+TypeScript compiler `tsc`. Hence, you get the same behaviour as if you had
+developed your code directly for Node.
+
+`deno2node` can perform more powerful transpilation steps that make it flexible
+enough for most needs.
 
 ### Shimming
 
-To use Deno globals not available in Node.js,
-create and register a file exporting all the shims you need:
+Some things are global in Deno, but not in Node.js. One example for this is
+`fetch`. Consequently, you need to _shim_ them, i.e. provide code that
+supplements the missing globals.
+
+For instance, you can use [`node-fetch`] as a substitue for the built-in `fetch`
+of Deno.
 
 ```sh
-$ npm install deno.ns
+npm install node-fetch
 ```
 
-```js
-// @filename: src/shim.node.ts
-export * from "deno.ns";
+Now, create a file that exports the globals you need:
+
+```ts
+// src/shim.node.ts
+export { default as fetch } from "node-fetch";
+
+// more shims exported here
 ```
 
+Lastly, you need to register your shims in `tsconfig.json` so `deno2node` can
+inject them for you:
+
 ```jsonc
-// @filename: tsconfig.json
 {
   "deno2node": {
     "shim": "src/shim.node.ts" // path to shim file, relative to tsconfig
   }
 }
 ```
 
+If you simply want to shim all Deno globals, you can install the `deno.ns`
+package. Then do:
+
+```ts
+// src/shim.node.ts
+export * from "deno.ns";
+```
+
+That package was specifically created for this use case.
+
 ### Runtime-specific code
 
-When the provided transformations are not enough,
-you can provide a Node-specific (`<anything>.node.ts`)
-and a Deno-specific (`<anything>.deno.ts`) version of any file.
+In same cases you may want to have two different implementations, depending on
+whether you're running on Deno or on Node. When shimming is not enough, you can
+provide a Node-specific `<anything>.node.ts` and a Deno-specific
+`<anything>.deno.ts` version of any file. They need to reside next to each other
+in the same directory.
+
+`deno2node` will ignore the Deno version and rewrite imports to use the Node.js
+version instead. Thus, the Deno-specific file will not be part of the build
+output.
+
+For example, provide `greet.deno.ts` for Deno:
+
+```ts
+// src/greet.deno.ts
+export function greet() {
+  console.log("Hello Deno!");
+  // access Deno-specific APIs here
+}
+```
+
+Now, provide `greet.node.ts` for Node:
+
+```ts
+// src/greet.node.ts
+export function greet() {
+  console.log("Hello Node!");
+  // access Node-specific APIs here
+}
+```
```

Comment:
This needs a better example. Like hashing!
https://github.com/fromdeno/cache/blob/8a9fcd45ec8d473734c592d26af65432034332bf/src/deps.deno.ts#L3-L5
https://github.com/fromdeno/cache/blob/8a9fcd45ec8d473734c592d26af65432034332bf/src/deps.node.ts#L3-L6

Reply:
The point is just that there can be platform-specific code, and this example is perfectly obvious. Despite `console.log` being identical.